some gui quirks

chelovechishko opened the pull request.

chelovechishko committed:
gui: newfriendlist: select row above when deleting profile

chelovechishko committed:
gui: chat: add power to copy person's id to clipboard via rmb

chelovechishko committed:
gui: chat: hide untouchable context menu actions for own identity

chelovechishko committed:
gui: make listwidget respect no_icons setting

chelovechishko committed:
gui: chatlobbywidget: not expand public 'chat rooms' anymore at program start

chelovechishko committed:
gui: make notify respect no_icons setting for listwidget

chelovechishko committed:
gui: statistics: add exit button onto toolbar
and comment out quit-close button per cyril taste

chelovechishko committed:
gui: statistics: set attribute Qt::WA_DeleteOnClose for the window

to free memory on close.

before opening->clicked through all tabs->after closing
without WA_DeleteOnClose:
 180->225->225 ; ->225->225...
with WA_DeleteOnClose:
 180->226->187 ; ->224->189...

There is a mud with spaceotabulation, so don't poke me for doing
automatic things for indentation, instead of understanding what should
be there for new code.
'tis no more saves last opened statistics' tab,
but hopefully nobody needs such a thing.

chelovechishko committed:
gui: chat: repudiate a hashbox to be half-screen large

csoler commented:
This PR seems to cause a crash: open the statistics, then close it using the Quit button. After 2 secs, the program crashes.

chelovechishko committed:
gui: statistics: comment out deleteonclose attribute

retroshare crashes

chelovechishko commented:
commented out wa_deleteonclose.
retroshare should crash at usual pace now.

chelovechishko committed:
gui: forum: set combobox property for versions of post

Only silhouette of word instead of time no more.
This is maximum length of time representation, but auto shrinkable.

chelovechishko committed:
gui: forum: not expand unread children no more

Make the setting great again

csoler commented:
- the PR still crashes when closing twice the statistics using the statistics button. Try with valgrind if it doesn't crash, valgrind will still tell you what's wrong. Or just cancel out that particular change: make the exit button to close the window instead of deleting it.
-

chelovechishko committed:
gui: chatlobbydialog: delete mparticipantcomparerole in destructor

chelovechishko committed:
gui: chatwidget: delete menus in destructor

As qt says: Ownership of the menu is not transferred to the tool button.

chelovechishko committed:
gui: rscombobox: add destructor

chelovechishko committed:
gui: settings: add missed ui deletion in postedpage

chelovechishko committed:
gui: statistics: delete bwdelegate in destructor

chelovechishko committed:
gui: rsgraphwidget: delete timer in destructor

chelovechishko committed:
gui: forum: add default destructor for itemdelegate and

delete delegatesforcolumn

chelovechishko committed:
gui: messagesdialog: delete itemdelegate in destructor

chelovechishko committed:
gui: iddialog: delete itemdelegate and menu in destructor

chelovechishko committed:
gui: grouptreewidget: delete comparerole and itemdelegate in destructor

chelovechishko committed:
gui: lineeditclear: add destructor

chelovechishko committed:
gui: mainwindow: delete idle in destructor

chelovechishko committed:
gui: sharedfilesdialog: add default destructor for shareflagsitemdelegate

chelovechishko committed:
gui: transfersdialog: delete also model in destructor

chelovechishko committed:
gui: homepage: delete also menu in destructor

chelovechishko committed:
gui: newsfeed: delete ui in destructor

chelovechishko committed:
gui: forum: delete threadtreewidget proxymodel in destructor

As qt says: the view does not take ownership of the model.

chelovechishko committed:
gui: messagewidget: delete menu in destructor

chelovechishko committed:
gui: rsimageblockwidget: delete menu in destructor

chelovechishko committed:
gui: handlerichtext: attempt to silence valgrind for qimage

"Conditional jump or move depends on uninitialised value"

chelovechishko committed:
gui: forum: set expanding sizepolicy for grouptreewidget

It still shrinks for some amount, but less.
What exactly cause such behaviour from post to post still unknown.

defnax reviewed (commented):
https://github.com/RetroShare/RetroShare/pull/2608/commits/47fc1902af44b6ea62b50137c2e313fc97b1fb21
This is a nice feature, would be great you do a short PR for this feature :)

chelovechishko commented:
Nah, it would be cumbersome. What for, if there is the pr already, and
it's even richer than just mere one commit ¯\_(ツ)_/¯
